hoodie-common/src/main/java/com/uber/hoodie/common/util/collection/DiskBasedMap.java

```diff
@@ -140,7 +142,8 @@ public void run() {
             }
           }
         } catch (Exception e) {
-          // fail silently for any sort of exception
+          // delete the file for any sort of exception
+          writeOnlyFile.delete();
```

Comment:
This is good, please add writeOnlyFile.delete() inside addShutDownHook() method as well.

Reply:
good point.